subsys/bluetooth/mesh/friend.c

```diff
@@ -873,8 +874,6 @@ static void enqueue_offer(struct bt_mesh_friend *frnd, int8_t rssi)
 		return;
 	}
 
-	frnd->counter++;
-
```

Comment:
Perhaps add a comment here explaining why we're not following spec and increasing the friend counter? In the added location, git blame can explain it, but here, it'll look like a bug.